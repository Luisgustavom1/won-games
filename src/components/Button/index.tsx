import { PropsWithChildren } from 'react';
import * as S from './styles';

export interface ButtonProps {
    $size?: 'small' | 'medium' | 'large';
    $fullWidth?: boolean;
}

export const Button = ({
    children,
    $fullWidth = false,
    $size = 'medium',
}: PropsWithChildren<ButtonProps>) => {
    return (
        <S.Wrapper $size={$size} $fullWidth={$fullWidth}>
            {!!children && <span>{children}</span>}
        </S.Wrapper>
    );
};
