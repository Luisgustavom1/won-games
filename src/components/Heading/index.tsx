import { ReactNode } from 'react';
import * as S from './styles';

export interface HeadingProps {
    children: ReactNode;
    color?: 'black' | 'white';
    $lineLeft?: boolean;
    $lineBottom?: boolean;
}

export const Heading = ({
    children,
    color = 'black',
    $lineLeft = false,
    $lineBottom = false,
}: HeadingProps) => {
    return (
        <S.Wrapper
            $lineLeft={$lineLeft}
            $lineBottom={$lineBottom}
            color={color}
        >
            {children}
        </S.Wrapper>
    );
};
