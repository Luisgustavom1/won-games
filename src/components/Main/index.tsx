import * as S from './style';

export const Main = ({
    title = 'Won games',
    description = 'Won games',
    primary = false,
    c = '#ffffff',
}) => {
    return (
        <S.Container data-testid="main">
            <h1 style={{ color: c }}>{title}</h1>
            <p>{description}</p>
            <code>{primary ? 'Primary' : 'Secondary'}</code>
        </S.Container>
    );
};
