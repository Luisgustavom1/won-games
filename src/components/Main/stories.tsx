import type { Meta, StoryObj } from '@storybook/react';
import { Main } from '.';

const meta: Meta<typeof Main> = {
    title: 'Main',
    component: Main,
    tags: ['autodocs'],
    args: {
        title: 'Won games default',
        description: 'Won games',
        primary: true,
    },
    argTypes: {
        c: {
            control: {
                type: 'color',
                presetColors: ['#000', '#fff'],
            },
            description: 'To set the color title',
            type: {
                name: 'string',
                required: true,
            },
        },
    },
};
export default meta;

type Story = StoryObj<typeof Main>;

export const Basic: Story = {};

export const Primary: Story = {
    args: {
        primary: true,
        title: 'Primary as true',
        description: 'Primary main',
    },
};

export const Secondary: Story = {
    args: {
        primary: false,
        title: 'Primary as false',
        description: 'Secondary main',
    },
};
