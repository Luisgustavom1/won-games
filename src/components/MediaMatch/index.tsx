import { css, styled } from 'styled-components';
import media, { DefaultBreakpoints } from 'styled-media-query';

type Breakpoint = keyof DefaultBreakpoints;

export interface MediaMatchProps {
    $lessThan?: Breakpoint;
    $greaterThan?: Breakpoint;
}

const mediaMathModifiers = {
    lessThan: (size: Breakpoint) => css`
        ${media.lessThan(size)`
            display: block
        `}
    `,

    greaterThan: (size: Breakpoint) => css`
        ${media.greaterThan(size)`
            display: block
        `}
    `,
};

export default styled.div<MediaMatchProps>`
    ${({ $lessThan, $greaterThan }) => css`
        display: none;

        ${!!$lessThan && mediaMathModifiers.lessThan($lessThan)}
        ${!!$greaterThan && mediaMathModifiers.greaterThan($greaterThan)}
    `}
`;
