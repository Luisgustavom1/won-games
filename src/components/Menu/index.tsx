import { Menu2 as MenuIcon } from '@styled-icons/remix-fill/Menu2';
import { ShoppingCart as ShoppingCartIcon } from '@styled-icons/material-outlined/ShoppingCart';
import { Search as SearchIcon } from '@styled-icons/material-outlined/Search';

import * as S from './styles';
import { Logo } from '../Logo';

export const Menu = () => {
    return (
        <S.Wrapper>
            <S.IconWrapper aria-label="Open Menu">
                <MenuIcon />
            </S.IconWrapper>
            <S.LogoWrapper>
                <Logo $hiddenOnMobile />
            </S.LogoWrapper>
            <S.MenuGroup>
                <S.IconWrapper aria-label='Search'>
                    <SearchIcon />
                </S.IconWrapper>
                <S.IconWrapper aria-label="Open Shopping Cart">
                    <ShoppingCartIcon />
                </S.IconWrapper>
            </S.MenuGroup>
        </S.Wrapper>
    );
};
