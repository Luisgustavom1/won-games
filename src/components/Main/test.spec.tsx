import { render, screen } from '@testing-library/react';
import { Main } from '@/components/Main';

describe('<Main />', () => {
    it('Should render heading', () => {
        const { container } = render(<Main />);

        expect(
            screen.getByRole('heading', { name: /won games/i }),
        ).toBeInTheDocument();
        expect(screen.getByTestId('main')).toHaveStyleRule('#123121');
        expect(container.firstChild).toMatchSnapshot();
    });
});
