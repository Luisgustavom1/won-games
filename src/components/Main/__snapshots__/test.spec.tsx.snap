// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Main /> > Should render heading 1`] = `
.c0 {
  display: flex;
  background-color: #123121;
  color: #919844;
}

<main
  class="c0"
  data-testid="main"
>
  <h1
    style="color: rgb(255, 255, 255);"
  >
    Won games
  </h1>
  <p>
    Won games
  </p>
  <code>
    Secondary
  </code>
</main>
`;
